src/main/java/picard/sam/SamToFastq.java

```diff
@@ -414,50 +435,46 @@ private void assertPairedMates(final SAMRecord record1, final SAMRecord record2)
      * messages to be written to the appropriate place.
      */
     protected String[] customCommandLineValidation() {
+
+        List<String> errors = new ArrayList<>();
+
         if (INTERLEAVE && SECOND_END_FASTQ != null) {
-            return new String[]{
-                    "Cannot set INTERLEAVE to true and pass in a SECOND_END_FASTQ"
-            };
+            errors.add("Cannot set INTERLEAVE to true and pass in a SECOND_END_FASTQ");
         }
 
         if (UNPAIRED_FASTQ != null && SECOND_END_FASTQ == null) {
-            return new String[]{
-                    "UNPAIRED_FASTQ may only be set when also emitting read1 and read2 fastqs (so SECOND_END_FASTQ must also be set)."
-            };
+            errors.add("UNPAIRED_FASTQ may only be set when also emitting read1 and read2 fastqs (so SECOND_END_FASTQ must also be set).");
         }
 
         if ((CLIPPING_ATTRIBUTE != null && CLIPPING_ACTION == null) ||
                 (CLIPPING_ATTRIBUTE == null && CLIPPING_ACTION != null)) {
-            return new String[]{
-                    "Both or neither of CLIPPING_ATTRIBUTE and CLIPPING_ACTION should be set."};
+            errors.add("Both or neither of CLIPPING_ATTRIBUTE and CLIPPING_ACTION should be set.");
         }
 
         if (CLIPPING_ACTION != null) {
-            if (CLIPPING_ACTION.equals("N") || CLIPPING_ACTION.equals("X")) {
-                // Do nothing, this is fine
-            } else {
+            if (!CLIPPING_ACTION.equals("N") && !CLIPPING_ACTION.equals("X")) {

```

Comment:
Should have a named constant for these strings as they're repeated a few times here. E.g. `CLIP_TRIM`, `CLIP_TO_N`.

Reply:
done